src/Stratis.Bitcoin.Features.Api/ApiFeature.cs

```diff
@@ -74,41 +88,19 @@ public override void Dispose()
                 this.webHost = null;
             }
         }
-
-        /// <summary>
-        /// A KeepaliveMonitor when enabled will shutdown the
-        /// node if no one is calling the keepalive endpoint
-        /// during a certain trashold window
-        /// </summary>
-        public void TryStartKeepaliveMonitor()
-        {
-            if (this.apiFeatureOptions.KeepaliveMonitor?.KeepaliveInterval.TotalSeconds > 0)
-            {
-                this.asyncLoop = this.asyncLoopFactory.Run("ApiFeature.KeepaliveMonitor", token =>
-                {
-                    // shortened for redability
-                    KeepaliveMonitor monitor = this.apiFeatureOptions.KeepaliveMonitor;
-
-                    // check the trashold to trigger a shutdown
-                    if (monitor.LastBeat.Add(monitor.KeepaliveInterval) < this.fullNode.DateTimeProvider.GetUtcNow())
-                        this.fullNode.Dispose();
-
-                    return Task.CompletedTask;
-                },
-                this.fullNode.NodeLifetime.ApplicationStopping,
-                repeatEvery: this.apiFeatureOptions.KeepaliveMonitor?.KeepaliveInterval,
-                startAfter: TimeSpans.Minute);
-            }
-        }
     }
 
     public sealed class ApiFeatureOptions
     {
-        public KeepaliveMonitor KeepaliveMonitor { get; private set; }
+        public Timer KeepaliveTimer { get; private set; }
 
         public void Keepalive(TimeSpan timeSpan)
         {
-            this.KeepaliveMonitor = new KeepaliveMonitor { KeepaliveInterval = timeSpan };
+            this.KeepaliveTimer = new Timer
+            {
+                AutoReset = false,
+                Interval = timeSpan.TotalSeconds * 1000
```

Comment:
Why the multiplication?
`timeSpan.TotalSeconds * 1000`

Reply:
because the interval takes in miliseconds.

Reply:
Ah in that case rename the method parameter to `keepaliveSeconds` thats how we do in other places.